Add JGroups DNS_PING protocol (from jgroups-kubernetes)

middagj opened the pull request:
I was missing the DNS_PING protocol. It is easier since you don't have to configure the additional privileges. This pull requests add this (only 1 jar extra will be added).

In JGroups 4 the DNS_PING protocol is included, but it was moved from jgroups-kubernetes pre 1.0. Since Artemis is stuck on JGroups 3, we can use the DNS_PING from jgroups-kubernetes 0.9.3 (the last compatible with JGroups 3).
See http://www.jgroups.org/manual4/index.html#_dns_ping

vromero commented:
Thanks. I'm currently on the road, I'll read carefully as soon as possible.

Off the top of my head I'm was not happy with the state of jgroups and artemis so far (see https://github.com/vromero/activemq-artemis-helm/tree/feature/kubeping) although it works due to the version of jgroups and the likeliness of jgroups3 kubeping geting behind the kubernetes API at any point in time.

I was planning to remove it, but perhaps this is the answer. I don't know yet.

middagj commented:
Did you have any time to read it? With DNS_PING you don't have API versioning issues, nor the authentication problems. In combination with the statefulset and the dns properties we have it works quite well.

vromero commented:
I'm not sure if you are aware I have a parallel project for a Helm chart for this docker image. Before merging this I'd like to try it there but I havent found the time for it yet. Also I'm not 100% convinced the jgroups path is the right one as at the end of the day an Artemis cluster is rather static.

middagj commented:
Yes, I am aware of the parallel project. See also my pull request to fix a typo: https://github.com/vromero/activemq-artemis-helm/pull/25, but with that correction it still does not work for me.

I like the JGroups for its flexibility to change replica count without having to release Helm. I don't think it would suffice for everyone, there are good arguments to use a static configuration. You already have JGroups libraries added to this project. I only request to add an additional library, so that I can use it in my own Helm chart. I think it benefits all if a single Docker image can be used in multiple ways.

vromero commented:
All right lets add it experimentally and lets see where it goes.
Could you please make sure tests pass correctly and that you update the documentation acordingly?

middagj commented:
Great! I rebased on master and added documentation. I don't know why the previous time the test failed, but it had something to do with the Alpine image, not the changes. All test passed this time.
